db/driver/driver.go

```diff
@@ -0,0 +1,87 @@
+package driver
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	"github.com/buildwithgrove/path/user"
+)
+
+// The PostgresDriver struct satisfies the Driver interface which defines all database driver methods
```

Comment:
Please enforce this via `var _  XXXInterface = &postgresDriver{}`

Reply:
Done. Also updated cache to accept driver as arg to avoid import cycle (this is a better design anyway).